firebase-test/firebasetest.js

```diff
@@ -10,18 +10,33 @@ var config = {
 
 firebase.initializeApp(config);
 
+var provider = new firebase.auth.GithubAuthProvider();
+
 // Create a JavaScript object for the HTML element that has id="message"
 var messageBox = document.getElementById("message");
 // Create a JavaScript object for the HTML element that has id="username"
 var usernameBox = document.getElementById("username");
 
+// Create Objects for Buttons UserInfo
+var userInfoBox = document.getElementById("userinfo");
+// Create Objects for Button Login
+var loginButton = document.getElementById("login");
+// Create Objects for Button Logout
+var logoutButton = document.getElementById("logout");
+
 // Get a reference to the root of our database
 var dbRef = firebase.database().ref();
 // Get a reference to the "greeting" section of our database
 var dbGreeting = dbRef.child("greeting");
 // Get a reference to the "myname" section of our database
 var dbUsername = dbRef.child("myname");
 
+// button function and event listener
+document.addEventListener("click", myButton);
```

Comment:
I'm a bit confused by this event listener -- did you get to test it? Seems like it wouldn't work right, or if this is leftover code from a temporary experiment, remember to delete it when you make your next edits. See my comments on line 54 and 62 below regarding the event listeners.